ebs/ebs_service.go

```diff
@@ -173,9 +175,14 @@ func (s *ebsService) CreateVolume(request *CreateEBSVolumeRequest) (string, erro
 		AvailabilityZone: aws.String(s.AvailabilityZone),
 		Size:             aws.Int64(ebsSize),
 	}
+
 	if snapshotID != "" {
 		params.SnapshotId = aws.String(snapshotID)
+	} else if kmsKeyID != "" {
+		params.KmsKeyId = aws.String(kmsKeyID)
+		params.Encrypted = aws.Bool(true)
 	}
+
 	if volumeType != "" {
 		if volumeType != "gp2" && volumeType != "io1" && volumeType != "standard" {
```

Comment:
hmm.. looks like i missed this conditional check on volume type re allowing st1, sc1.